homeassistant/components/brunt/cover.py

```diff
@@ -153,25 +169,46 @@ def is_closed(self):
         """Return true if cover is closed, else False."""
         return self.current_cover_position == CLOSED_POSITION
 
-    def update(self):
-        """Poll the current state of the device."""
-        try:
-            self._state = self._bapi.getState(thingUri=self._thing_uri).get("thing")
-            self._available = True
-        except (TypeError, KeyError, NameError, ValueError) as ex:
-            _LOGGER.error("%s", ex)
-            self._available = False
-
-    def open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs):
         """Set the cover to the open position."""
-        self._bapi.changeRequestPosition(OPEN_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(OPEN_POSITION)
 
-    def close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs):
         """Set the cover to the closed position."""
-        self._bapi.changeRequestPosition(CLOSED_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(CLOSED_POSITION)
 
-    def set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs):
         """Set the cover to a specific position."""
-        self._bapi.changeRequestPosition(
-            kwargs[ATTR_POSITION], thingUri=self._thing_uri
+        await self._async_update_cover(kwargs[ATTR_POSITION])
+
+    async def _async_update_cover(self, position):
+        """Set the cover to the new position and wait for the update to be reflected."""
+        await self._bapi.async_change_request_position(
+            position, thingUri=self._thing.thingUri
         )
+        self.coordinator.update_interval = FAST_INTERVAL
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _brunt_update_listener(self):
+        """Update the update interval after each refresh."""
```

Comment:
How would this work with two covers retrieving their latest state? I would assume that it should switch to `FAST_INTERVAL` when no devices are waiting for an update, but I am not sure if this currently works since you do this from within each entity.

Reply:
Since the dataupdatecoordinator updates all states at once, this is not an issue, as long as 1 cover is updating, it stays on FAST, otherwise returns to REGULAR.